core/kazoo_number_manager/src/knm_phone_number.erl

```diff
@@ -256,27 +268,38 @@ handle_bulk_change(Db, JObjs, PNsMap, T0, ErrorF)
                         ErrorF(Num, kz_term:to_atom(R, 'true'), T)
                 end
         end,
-    retry_conflicts(lists:foldl(F, T0, JObjs), Db, PNsMap, ErrorF);
-handle_bulk_change(Db, JObjs, PNs, T, ErrorF) ->
+    retry_conflicts(lists:foldl(F, T0, JObjs), Db, PNsMap, RetryF);
+handle_bulk_change(Db, JObjs, PNs, T, ErrorF, RetryF) ->
     PNsMap = group_by_num(PNs),
-    handle_bulk_change(Db, JObjs, PNsMap, T, ErrorF).
+    handle_bulk_change(Db, JObjs, PNsMap, T, ErrorF, RetryF).
 
-handle_bulk_change(Db, JObjs, PNs, T) ->
-    ErrorF = fun database_error/3,
-    handle_bulk_change(Db, JObjs, PNs, T, ErrorF).
+handle_bulk_change(Db, JObjs, PNs, T, ErrorF) ->
+    RetryF = fun(Db1, PNs1, Num, T1) ->
+                     retry_save(Db1, PNs1, ErrorF, Num, T1)
+             end,
+    handle_bulk_change(Db, JObjs, PNs, T, ErrorF, RetryF).
+
+-spec retry_save(kz_term:ne_binary(), pns_map(), bulk_change_error_fun(), knm_numbers:num(), knm_numbers:collection()) ->
+                        knm_numbers:collection().
+retry_save(Db, PNsMap, ErrorF, Num, T) ->
+    PN = maps:get(Num, PNsMap),
+    case kz_datamgr:ensure_saved(Db, to_json(PN)) of
```

Comment:
Hi @danielfinke sorry this has languished so long. Can you remove the use of the deprecated `ensure_saved` in favor of `update_doc`? Then I will merge this

Reply:
@jamesaimonetti gonna switch it to `save_doc/2` instead unless there's a reason I'm missing as to why you suggested `update_doc`?

Reply:
Only that update_doc has an ensure_saved flag

Reply:
@jamesaimonetti ok, using update_doc/3 instead

Reply:
@jamesaimonetti hey, anything else needed from your end to get this thru?